utils/zerovec/src/zerovec/crabbake.rs

```diff
@@ -0,0 +1,25 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::ZeroVec;
+use crate::ule::AsULE;
+use crabbake::*;
+
+impl<T> Bakeable for ZeroVec<'_, T>
+where
+    T: AsULE + ?Sized,
+{
+    fn bake(&self, env: &CrateEnv) -> TokenStream {
+        env.insert("core");
+        env.insert("zerovec");
+        let bytes = self.as_bytes();
+        let size: usize = core::mem::size_of::<T::ULE>();
+        quote! { unsafe {
+            static DATA: &[u8] = &[#(#bytes),*];
+            let (data, mut metadata): (usize, usize) = core::mem::transmute(DATA);
+            metadata /= #size;
+            zerovec::ZeroVec::Borrowed(core::mem::transmute((data, metadata)))
+        }}
```

Comment:
Nit: The unsafeness of this code is very high. Are you sure there's no other way to do it?

CC @Manishearth

Reply:
Yes

Reply:
Yes, until some more things become const. I think we'll be able to at least make this a method on ZeroVec once current nightly becomes stable.

Reply:
Fixed by #1926